web/film-club-client/src/components/FilmDetailPage.js

```diff
@@ -1,54 +1,193 @@
 
 import Header from './Header';
 
-const FilmDetailPage = props => {
+const FilmDetailPage = (props) => {
+	const [movieDetails, setMovieDetails] = useState({});
+	const [showVideo, setShowVideo] = useState(false);
+
 	// React Router URL Parameter object
 	let { id } = useParams();
 
-	// const getMovieDetails = async (id) => {
-	//     const response = await fetch(`http://localhost:8080/v1/films/${id}`)
-	//     const data = await response
-	//     // props.setMovieId(data)
-	//     console.log(data)
-	// }
-
-	// getMovieDetails(id)
+	// // Fetch all movie details with {id} from params, on component load
+	useEffect(() => {
+		// Use the {id} to fetch details about the corresponding movie
+		const getMovieDetails = async (id) => {
+			const response = await fetch(`http://localhost:8080/v1/films/${id}`);
+			const data = await response.json();
+			console.log(data);
+			setMovieDetails(data);
+		};
+		getMovieDetails(id);
+	}, [id]);
 
-	let { posterPath, title, overview } = props.movieId;
+	let {
+		backdropPath,
+		credits,
+		images,
+		genres,
+		posterPath,
+		releaseDate,
+		similar,
+		tagline,
+		title,
+		overview,
+		videos,
+		watchProviders,
+	} = movieDetails;
 
-	return (
-		<Box>
-			<Header />
-			{id ? (
+	if (movieDetails.images) {
+		return (
+			<Box
+				className="page-box"
+				display="flex"
+				direction="column"
+				justify="around"
+				background="dark-2"
+				height={{ min: '100vh' }}
+				overflow={{
+					vertical: 'scroll',
+				}}
+				pad="medium"
+			>
+				<Header />
 				<Box
+					className="hero-box"
 					display="flex"
-					direction="column"
+					direction="row"
 					justify="around"
-					align="center"
-					background="dark-2"
-					pad={{ vertical: '6em', horizontal: '5em' }}
-					height="auto"
+					// align="center"
+					margin={{ top: '4em' }}
 				>
-					<Box pad="small">
-						<Image
-							src={`http://localhost:8080/v1/films/images/poster${posterPath}?size=medium`}
-							alt={`${title} poster image`}
-						></Image>
+					<Box
+						className="poster-box"
+						pad={{ right: 'medium' }}
+						width={{ min: '5em' }}
+					>
+						<Box>
+							<Image
+								src={`http://localhost:8080/v1/films/images/poster${posterPath}?size=large`}
+								alt={`${title} poster image`}
+							/>
+						</Box>
+						<Box
+							className="trailer-box"
+							display="flex"
+							direction="row"
+							justify="around"
+							onClick={() => setShowVideo(true)}
+							focusIndicator={false}
+							margin={{ top: 'small' }}
+							pad="small"
+							border={{ size: 'small', color: 'accent-4' }}
+							round="small"
+						>
+							<Text>Watch Trailer</Text>
+							<CirclePlay size="25em" color="accent-4" />
+						</Box>
+						{showVideo && (
+							<Layer
+								modal={true}
+								responsive={true}
+								onEsc={() => setShowVideo(false)}
+								onClickOutside={() => setShowVideo(false)}
+								round="medium"
+							>
+								{videos[0] ? (
```

Comment:
The first video is not guaranteed to be a `Trailer`! Maybe you can search the videos to [find the first one](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) that's a `Trailer`?